File: Dataset/Leetcode/train/62/500.py
class Solution:
    def XXX(self, m: int, n: int) -> int:
        a=[];b=[]
        for i in range(n):
            a.append(1)
        for j in range(m):
            b.append(a)
        for i in range(1,m):
            for j in range(1,n):
                b[i][j]=b[i-1][j]+b[i][j-1]
        return b[m-1][n-1]

File: Dataset/Leetcode/test/125/500.cpp
 class Solution {
public:
    bool XXX(string s) {
        int pos1 = 0, pos2 = s.size() - 1; //双指针遍历
        while(pos1 < pos2){
            char front = s[pos1];
            char rear = s[pos2];
            int flag = 0;
            if(!isalnum(front)){
                pos1++;
                flag = 1;
            } 
            if(!isalnum(rear)){
                pos2--;
                flag = 1;
            } 
            
            if(flag) continue;
            if(front-rear==0 || tolower(front)-tolower(rear)==0){
                pos1++; pos2--;
            }else{
                return false;
            }   
        }
        return true;
    }
};

